src/lib/y2s390/zfcp.rb

```diff
@@ -90,6 +113,20 @@ def activated_controller?(channel)
       io.any? { |i| i["active"] }
     end
 
+    # Whether the controller is performing auto LUN scan
+    #
+    # For that, allow_lun_scan must be active and the controller must be running in NPIV mode.
+    #
+    # @param channel [String] E.g., "0.0.fa00"
+    # @return [Boolean]
+    def lun_scan_controller?(channel)
+      file = "/sys/bus/ccw/drivers/zfcp/#{channel}/host0/fc_host/host0/port_type"
```

Comment:
I'm hardly familiar with YaST and ruby, so I'm not sure I fully understand what the code does.
"host0" looks hard-coded but the actual number of the Scsi_Host / fc_host is not fixed or pre-determined. The number gets assigned by the Linux SCSI stack during registration of a zfcp device (what the code calls channel, but is strictly speaking the device-bus ID for a *sub*channel representing a hardware-virtualized HBA).
Since there should only ever be exactly one associated Scsi_Host / fc_host per "#{channel}" it would work to replace "host0" with "host*" and let file glob pattern expansion expand the file path to whatever fc_host the thing might have at runtime.
Regards, a zfcp maintainer.